src/lib.nix

```diff
@@ -71,6 +71,24 @@
   }:
     initDream2nix (loadConfig config) pkgs;
 
+  missingProjectsError = source: ''
+    Please pass `projects` to makeFlakeOutputs.
+    `projects` can be:
+      - an attrset
+      - a path to a .toml file (not empty & added to git)
+      - a path to a .json file (not empty & added to git)
+
+    To generate a projects.toml file automatically:
+      1. execute:
+        nix run github:nix-community/dream2nix#detect-projects ${source} > projects.toml
+      2. review the ./projects.toml and edit it if necessary.
+      3. pass `projects = ./projects.toml` to makeFlakeOutputs.
```

Comment:
I think instead of running `detect-projects` from an external dream2nix we should expose an app called `detect-projects` under outputs. The main dream2nix repo here may not have the discovers / translators / builders etc. a user may add which would cause projects to not get detected etc.

Reply:
That doesn't address the issue, because modules are added in the flake itself. So the modules would still be unloaded for that particular instance of dream2nix that would be used. I think adding an app would be fine, since even if there is an issue somewhere else in the flake itself, running `nix run .#detect-projects` will still work regardless because of lazyness.

Reply:
> I thought about that, but decided against it, because the flake itself might crash because of some mistake in the toml or flake and then the app won't be available.

What kind of mistakes are we talking about here? 

Malformed TOML files seem to produce relatively specific errors (albeit without the files path ofc because fromTOML doesn't know that), e.g. `error: while parsing a TOML string: [...]`.

Configuration errors are a much more open problem, but not specific to `detect-projects`, unless I am missing something? 

But on the other hand, @yusdacra, wouldn't your use case be covered by using `nix run path-to-local-dream2nix-checkout#detect-projects` or is this a documentation issue?

Reply:
> wouldn't your use case be covered by using nix run path-to-local-dream2nix-checkout#detect-projects or is this a documentation issue?

No, the issue here is that a user may initialize a dream2nix instance with their own modules, in their `flake.nix` file. This will be specific to *only* that instance, and not any other instance, `nix run dream2nix-checkout#detect-projects` will not be aware of this. And so if new discoverers / translators etc. are added in the user's instance, the `nix run checkout#detect-projects` will not detect the projects that the new discoverers would, because that instance will not have the modules added.

Reply:
resolved by f83a752d38e95e4748b5b672f19b4ab4bddb8cc8